src/modules/claim/resolvers/credential.resolver.ts

```diff
@@ -1,69 +1,104 @@
-import jsonwebtoken from 'jsonwebtoken';
 import {
   CredentialResolver,
+  isEIP191Jwt,
+  isVerifiableCredential,
   RoleEIP191JWT,
+  transformClaim,
   VerifiableCredential,
 } from '@energyweb/vc-verification';
-import { ClaimData } from '@energyweb/vc-verification/dist/src/models';
+import { RolePayload } from '@energyweb/vc-verification';
 import { RoleCredentialSubject } from '@energyweb/credential-governance';
-import { IPublicClaim } from '@ew-did-registry/claims';
-import { ClaimService } from '../services';
-import { RegistrationTypes } from '../claim.types';
+import * as jwt from 'jsonwebtoken';
+import { DIDService } from '../../did/did.service';
 
 export class RoleCredentialResolver implements CredentialResolver {
-  constructor(private readonly claimService: ClaimService) {}
+  constructor(private readonly didService: DIDService) {}
+
   async getCredential(
     did: string,
     namespace: string
   ): Promise<
     VerifiableCredential<RoleCredentialSubject> | RoleEIP191JWT | null
   > {
+    const tokens = await this.didService.resolveServiceEndpoints(did);
```

Comment:
IMO, this is a bit clearer of a name. didDocument service endpoints don't necessarily resolve to a `token` so it's a bit unintuitive that this is the name.
```suggestion
    const resolvedEndpoints = await this.didService.resolveServiceEndpoints(did);
```